pkg/extensions/tbtc/tbtc.go

```diff
@@ -934,31 +938,31 @@ func (t *tbtc) waitDepositStateChangeConfirmation(
 }
 
 func (t *tbtc) waitKeepNotActiveConfirmation(
-	keepAddress string,
+	keep chain.BondedECDSAKeepHandle,
 ) bool {
-	currentBlock, err := t.chain.BlockCounter().CurrentBlock()
+	currentBlock, err := t.blockCounter.CurrentBlock()
 	if err != nil {
 		logger.Errorf(
 			"could not get current block while confirming "+
 				"keep [%v] is not active: [%v]",
-			keepAddress,
+			keep.ID(),
 			err,
 		)
 		return false
 	}
 
 	isKeepActive, err := ethlike.WaitForBlockConfirmations(
-		t.chain.BlockCounter(),
+		t.blockCounter,
 		currentBlock,
 		t.blockConfirmations,
 		func() (bool, error) {
-			return t.chain.IsActive(common.HexToAddress(keepAddress))
+			return keep.IsActive()
 		},
 	)
 	if err != nil {
 		logger.Errorf(
 			"could not confirm if keep [%v] is not active: [%v]",
-			keepAddress,
+			keep.ID(),
```

Comment:
And here

```suggestion
			keep.ID().String(),
```

Reply:
This is a known problem, we discussed it in the previous PR: https://github.com/keep-network/keep-ecdsa/pull/709

We decided to go ahead and merge because we'll refactor the ID later this week anyway.

Reply:
Got it, thanks! I mark this as resolved then.